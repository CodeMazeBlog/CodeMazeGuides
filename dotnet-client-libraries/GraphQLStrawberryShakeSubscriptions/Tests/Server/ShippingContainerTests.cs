using FluentAssertions;
using Snapshooter.Xunit;

namespace Tests.Server;
public class ShippingContainerTests
{
    [Fact]
    public async Task GivenShippingContainerSchema_WhenChanged_ThenCompareToSnapshot()
    {
        var schema = await TestServices.Executor.GetSchemaAsync(default);
        schema.ToString().Should().MatchSnapshot();
    }

    [Fact]
    public async Task GivenShippingContainer_WhenAddAvaliableShippingContainerAsync_ThenCompareToSnapshot()
    {
        var result = await TestServices.ExecuteRequestAsync(
            b => b.SetQuery(@"
                mutation AddAvaliableShippingContainer {
                    addAvaliableShippingContainer(input: {
                        name: ""Container 1"",
                        length: 10,
                        width: 10,
                        height: 10
                    }) 
                    {
                        shippingContainer {
                            id
                            name
                            space {
                                length
                                width
                                height
                            }
                        }
                    }
                }"));

        result.Should().MatchSnapshot();
    }

    [Fact]
    public async Task GivenShippingContainer_WhenUpdateShippingContainerAsync_ThenCompareToSnapshot()
    {
        await TestServices.ExecuteRequestAsync(
            b => b.SetQuery(@"
                mutation AddAvaliableShippingContainer {
                    addAvaliableShippingContainer(input: {
                        name: ""Container 2"",
                        length: 10,
                        width: 10,
                        height: 10
                    }) 
                    {
                        shippingContainer {
                            id
                            name
                            space {
                                length
                                width
                                height
                            }
                        }
                    }
                }"));

        var result = await TestServices.ExecuteRequestAsync(
            b => b.SetQuery(@"
                mutation UpdateShippingContainer {
                    updateShippingContainer(input: {
                        name: ""Container 2"",
                        length: 20,
                        width: 20,
                        height: 20
                    }) 
                    {
                        shippingContainer {
                            id
                            name
                            space {
                                length
                                width
                                height
                            }
                        }
                    }
                }"));

        result.Should().MatchSnapshot();
    }

    [Fact]
    public async Task GivenShippingContainer_WhenGetShippingContainersAsync_ThenCompareToSnapshot()
    {
        await TestServices.ExecuteRequestAsync(
            b => b.SetQuery(@"
                mutation AddAvaliableShippingContainer {
                    addAvaliableShippingContainer(input: {
                        name: ""Container 3"",
                        length: 10,
                        width: 10,
                        height: 10
                    }) 
                    {
                        shippingContainer {
                            id
                            name
                            space {
                                length
                                width
                                height
                            }
                        }
                    }
                }"));

        var result = await TestServices.ExecuteRequestAsync(
            b => b.SetQuery(@"
                query GetShippingContainers {
                    shippingContainers {
                        id
                        name
                        space {
                            length
                            width
                            height
                        }
                    }
                }"));

        result.Should().MatchSnapshot();
    }
}