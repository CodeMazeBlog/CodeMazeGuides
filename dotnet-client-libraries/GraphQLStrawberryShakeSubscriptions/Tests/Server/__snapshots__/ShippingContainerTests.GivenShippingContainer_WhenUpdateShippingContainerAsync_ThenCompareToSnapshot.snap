{
  "data": {
    "updateShippingContainer": {
      "shippingContainer": {
        "id": "0",
        "name": "Container 2",
        "space": {
          "length": 20,
          "width": 20,
          "height": 20
        }
      }
    }
  }
}
